### ISSUE TITLE
[Inductor] [CPU] Accuracy failure in torchbench model hf_BigBird

### ISSUE THREAD

**Issue by ESI-SYD** [pytorch/pytorch#94268]
### 🐛 Describe the bug

Error:

```
cpu  eval  hf_BigBird                          [2023-02-06 06:58:34,278] torch._dynamo.utils: [ERROR] RMSE (res-fp64): 0.20305, (ref-fp64): 0.00000 and shape=torch.Size([4, 819, 50358])
[2023-02-06 06:58:34,279] torch._dynamo.utils: [ERROR] Accuracy failed for key name logits
FAIL
```

WW06.2 SW info:

SW	| Nightly commit	| Master/Main commit
-- | -- | --
Pytorch|[6a03ad6](https://github.com/pytorch/pytorch/commit/6a03ad6a34d7c2a31fcaab7a28898f1777baac85)|[1d53123](https://github.com/pytorch/pytorch/commit/1d53123f44e2d5f08e4605af353b7d32b62346ae)
Torchbench|/|[ef39f8c](https://github.com/pytorch/benchmark/commit/ef39f8ca47e288ce7206fe92a0eaac03e85623b8)
torchaudio|[ecc2781](https://github.com/pytorch/audio/commit/ecc2781b5b5c353e85a8ca9f16741cf92dd08344)|[4a037b0](https://github.com/pytorch/audio/commit/4a037b03915c4f6f81407e697fb11a7e7ace27fa)
torchtext|[112d757](https://github.com/pytorch/text/commit/112d757efd2482ea5f60bc23d85c2c09f0cfde61)	| [c7cc5fc](https://github.com/pytorch/text/commit/c7cc5fc1e669f548eef619ae055690831d6cc75e)
torchvision|[ac06efe](https://github.com/pytorch/vision/commit/ac06efed4ad2867c68e6b49ac31b554fcbbaa472)|[35f68a0](https://github.com/pytorch/vision/commit/35f68a09f94b2d7afb3f6adc2ba850216413f28e)
torchdata|[049fb62](https://github.com/pytorch/data/commit/049fb626615e6cd965897af8fea2fa73cecd6a2a)|[c0934b9](https://github.com/pytorch/data/commit/c0934b9afa96458c0f8aa3bef528830835c22195)
dynamo_benchmarks|[6a03ad6](https://github.com/pytorch/pytorch/commit/6a03ad6a34d7c2a31fcaab7a28898f1777baac85)|/

### Versions

Repro CMD:

```
python -m torch.backends.xeon.run_cpu --node_id 0 benchmarks/dynamo/torchbench.py --accuracy --float32 -dcpu --inductor  --no-skip --dashboard --only hf_BigBird

```

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh

**Comment by chunyuan-w** [comment]
Analysis from @blzheng: 
- The change of the function `guard_equals` in https://github.com/pytorch/pytorch/pull/92609 made the accuracy of this model fail.

- When creating `View`, the `try` block should raise assertion when doing `guard_equals` while the current code does not raise assertion anymore.
  https://github.com/pytorch/pytorch/blob/a81cf49d9733b04a2931c85a154ab0bb698650b3/torch/_inductor/ir.py#L1331-L1340

I've drafted https://github.com/pytorch/pytorch/pull/94506 to fix the above issue.
Note that https://github.com/pytorch/pytorch/pull/90150 is also needed to run the `hf_BigBird` model.
